### PR TITLE
Add settings and isa command-line options to cton-util wasm.

### PR DESCRIPTION
**Pull request by sunfishcode**
This implements the full parsing for `set` and `isa` commands, following the [feedback](https://github.com/stoklund/cretonne/pull/155#issuecomment-329295426).

### CODE REVIEW

File: src/wasm.rs [hunk 1]
```diff
@@ -50,16 +53,32 @@ pub fn run(
...
+        None => return Err(String::from("expected ISA name")),
+        Some(w) => w,
+    };
+    let isa_builder = match isa::lookup(isa_name) {
+        Err(isa::LookupError::Unknown) => return Err(format!("unknown ISA '{}'", isa_name)),
+        Err(isa::LookupError::Unsupported) => {
+            return Err(format!("support for ISA '{}' not enabled", isa_name))
+        }
+        Ok(b) => b,
+    };
```
Comment (stoklund) [inline src/wasm.rs]: This might work better with `map_err` combined with a `?` operator, but NBD.

File: src/wasm.rs [hunk 2]
```diff
@@ -70,6 +89,7 @@ pub fn run(
             path.to_path_buf(),
             name,
             &flags,
+            isa.borrow(),
```
Comment (stoklund) [inline src/wasm.rs]: The good old C++ trick `&*isa` works too.

File: src/wasm.rs [hunk 3]
```diff
@@ -50,16 +53,32 @@ pub fn run(
...
+        flag_set.iter().map(|x| x.as_str()),
+        &mut flag_builder,
+        &Location { line_number: 0 },
+    ).map_err(|err| err.to_string())?;
     let flags = settings::Flags::new(&flag_builder);
 
+    let mut words = flag_isa.trim().split_whitespace();
+    // Look for `isa foo`.
+    let isa_name = match words.next() {
+        None => return Err(String::from("expected ISA name")),
```
Comment (stoklund) [inline src/wasm.rs]: Does this mean that `cton-util wasm` doesn't work without an `--isa` option? Was that intended?

Comment (sunfishcode) [reply src/wasm.rs]: Yes, for the moment. I'll submit a PR to create a host-detection crate and use that for a default soon.

**Pull request merged**